Win 10 pro still getting `EMFILE: too many open files` error.

I've added this at the root of my app

```
var realFs = require('fs')
var gracefulFs = require('graceful-fs')
gracefulFs.gracefulify(realFs)
```

But I'm still getting `EMFILE: too many open files` error. 

Any ideas what else can I do? Win 10 pro

@strarsis: I've "globally" patched fs in my index.js, the first line that's executed is the patch:
```
const realFs = require('fs');
// eslint-disable-next-line import/newline-after-import
const gracefulFs = require('graceful-fs');
gracefulFs.gracefulify(realFs);
const http = require('http');
const socketio = require('socket.io');
more requires and business logic...
```

I've manually patched the `fs` import in Handler.js as well:
```
const Promise = require('bluebird');
const fs = Promise.promisifyAll(require('graceful-fs'));
...
```

However, weirdly enough the error is thrown from exactly the same line of code like before; Handler:107. Here's the stack trace:
```
Extracted file with content <Buffer 89 50 4e 47 0d 0a 1a 0a 00 00 00 0d 49 48 44 52 00 00 01 00 00 00 00 c0 08 06 00 00 00 eb ed bd 66 00 00 c2 50 49 44 41 54 78 9c ec fd 75 bc 65 57 7d ... 49751 more bytes>
node:fs:495
  handleErrorFromBinding(ctx);
  ^

Error: EMFILE: too many open files, open 'data/resource/flash/econ/stickers/cologne2015/sig_kennys_gold_large.png'
    at Object.openSync (node:fs:495:3)
    at Object.writeFileSync (node:fs:1519:35)
    at ExtractSticker.extractFile (C:\Users\known\Desktop\ChatChops\alert\src\services\Extractors\Handler.js:107:7)
    at ExtractSticker.execute (C:\Users\known\Desktop\ChatChops\alert\src\services\Extractors\ExtractSticker.js:43:9)
    at SteamAssetExtractor.extractAssets (C:\Users\known\Desktop\ChatChops\alert\src\services\SteamAssetExtractor.js:104:17)
    at new SteamAssetExtractor (C:\Users\known\Desktop\ChatChops\alert\src\services\SteamAssetExtractor.js:27:8)
    at SteamAssetDownloader.<anonymous> (C:\Users\known\Desktop\ChatChops\alert\src\index.js:59:25)
    at SteamAssetDownloader.emit (node:events:376:20)
    at SteamAssetDownloader.set ready [as ready] (C:\Users\known\Desktop\ChatChops\alert\src\services\SteamAssetDownloader.js:33:9)
    at SteamAssetDownloader.update (C:\Users\known\Desktop\ChatChops\alert\src\services\SteamAssetDownloader.js:132:14) {
  errno: -4066,
  syscall: 'open',
  code: 'EMFILE',
  path: 'data/resource/flash/econ/stickers/cologne2015/sig_kennys_gold_large.png'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! alert@1.0.0 start: `node src/index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the alert@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\known\AppData\Roaming\npm-cache\_logs\2020-12-30T15_49_41_183Z-debug.log
```

--- a/README.md
+++ b/README.md
@@ -30,8 +30,18 @@ the directory.
 var fs = require('graceful-fs')
 
 // now go and do stuff with it...
-fs.readFileSync('some-file-or-whatever')
+fs.readFile('some-file-or-whatever', (err, data) => {
+  // Do stuff here.
+})
 ```
+
+## Sync methods
+
+This module cannot intercept or handle `EMFILE` or `ENFILE` errors from sync
+methods.  If you use sync methods which open file descriptors then you are
+responsible for dealing with any errors.
+
+This is a known limitation, not a bug.
 
 ## Global Patching
 

